### PR TITLE
9-implementing edit feature

### LINKED ISSUE #9: Edit Function

### PR DESCRIPTION
**Pull request by Ibrahim-Jarada**
Relates #9

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -6,15 +6,20 @@ import './App.css';
...
   const items = useMemo(
-    () => itemsArray.map((item) => (
+    () => itemsArray.map((item, index) => (
       <li
         className="list-group-item d-flex justify-content-between align-items-center"
         key={item.id}
       >
         <span className="title">{item.task}</span>
         <span>
+          <button type="button" onClick={() => handelEdit(index)} data-edit>
```
Comment (Jood80) [inline src/App.jsx:20]: why is the need to define an `index` when you already have an `id` in your `item`?

File: src/App.jsx [hunk 2]
```diff
@@ -30,21 +35,37 @@ Itemslist.propTypes = {
   className: PropTypes.string.isRequired,
   list: PropTypes.arrayOf(PropTypes.string).isRequired,
   handelDelete: PropTypes.func.isRequired,
+  handelEdit: PropTypes.func.isRequired,
 };
 
 function App() {
   const [itemsArray, setitemsArray] = useState([]);
   const [inputvalue, setInputvalue] = useState('');
```
Comment (Jood80) [inline src/App.jsx:43]: name your variables/functions consistently, as you've chosen the `camelCase` as a convention here stick with it

File: src/App.jsx [hunk 3]
```diff
@@ -30,21 +35,37 @@ Itemslist.propTypes = {
...
+  const [btnFunction, setBtnFunction] = useState('newItem');
+  const [editindex, setEditindex] = useState(-1);
```
Comment (Jood80) [inline src/App.jsx:45]: you don't need to define a new state for picking the index of your tasks, it's simply there in the id of `itemsArray` itself

File: src/App.jsx [hunk 4]
```diff
@@ -56,7 +77,7 @@ function App() {
           placeholder="Add task..."
           onChange={handleChange}
```
Comment (Jood80) [inline src/App.jsx:82]: At your current code, after submitting you'll still have the input value shine in your input field, and why is that?
have you made sure that you emptied the `inputValue` at the end of your `handleSubmit` function & set a `value` for your `<input .....` field?

File: src/App.jsx [hunk 5]
```diff
@@ -30,21 +35,37 @@ Itemslist.propTypes = {
   className: PropTypes.string.isRequired,
   list: PropTypes.arrayOf(PropTypes.string).isRequired,
```
Comment (Jood80) [inline src/App.jsx]: you sure it's an array of `string`? not an`object`?

File: src/App.jsx (hunk 1 above)
Comment (Ibrahim-Jarada) [reply src/App.jsx:20]: I don't want to search for index by ID when I can simply have the index directly
like this itemsArray.findIndex(item=>item.id===idValue)

**Pull request merged**